### PR TITLE
test: add unit tests for all conservation rules

### LINKED ISSUE #268: Write more unit tests for conservation_rules module
Currently not covered by the unit tests are:

- [x] `ParityConservation`
- [x] `ParityConservationHelicity`
- [x] `IdenticalParticleSymmetrization`
- [x] `ClebschGordanCheckHelicityToCanonical`
- [x] `HelicityConservation`
- [x] `GellMannNishijimaRule`

### PR DESCRIPTION
**Pull request by spflueger**
Closes #268 

Note that this may in the future avoid issues like #376

### CODE REVIEW

File: src/expertsystem/reaction/conservation_rules.py [hunk 1]
```diff
@@ -382,21 +385,20 @@ def check_g_parity_isobar(
...
     pid: EdgeQuantumNumbers.pid = attr.ib()
 
 
 def identical_particle_symmetrization(
-    ingoing_edge_qns: List[IdenticalParticleSymmetryEdgeInput],
-    outgoing_edge_qns: List[IdenticalParticleSymmetryEdgeInput],
+    ingoing_parities: List[EdgeQuantumNumbers.parity],
+    outgoing_edge_qns: List[IdenticalParticleSymmetryOutEdgeInput],
 ) -> bool:
     """Implementation of particle symmetrization."""
```
Comment (redeboer) [inline src/expertsystem/reaction/conservation_rules.py:398]: What exactly is particle symmetrization?

**Review by redeboer — APPROVED** [review]
Looks good!

Some general thoughts:
- I wonder if in the long term it makes sense to write some notebooks on how to use conservation rules. Ideally, these should work standalone, right?
- All the separate test files under
  `tests/unit/reaction/conservation_rules`
  could ([should](https://pwa.readthedocs.io/projects/expertsystem/en/0.6.3/develop.html#testing)) be merged under
  `tests/unit/reaction/test_conservation_rules`

**Comment by spflueger** [comment]
> * I wonder if in the long term it makes sense to write some notebooks on how to use conservation rules. Ideally, these should work standalone, right?

I don't think anyone will use those rules separately. If this will be the case for some reason. Then we can add some docs for that. Atm I do not see the need for that.

> * All the separate test files under
>   `tests/unit/reaction/conservation_rules`
>   could ([should](https://pwa.readthedocs.io/projects/expertsystem/en/0.6.3/develop.html#testing)) be merged under
>   `tests/unit/reaction/test_conservation_rules`

You mean to avoid that extra package layer and get a more fine grained test output log?

File: src/expertsystem/reaction/conservation_rules.py (hunk 1 above)
Comment (spflueger) [reply src/expertsystem/reaction/conservation_rules.py:398]: I can write a bit more detailed description

**Comment by redeboer** [comment]
> You mean to avoid that extra package layer and get a more fine grained test output log?

In the 'develop page', I wrote a note that the structure of unit test should reflect that of the components they test. In this case, the module `conservation_rules` should get a test called `test_conservation_rules` and its functions (say, `identical_particle_symmetrization`), should be named accordingly (`test_identical_particle_symmetrization`). But it's a debatable convention and it's not enforced through style checks or so.

**Comment by spflueger** [comment]
> > You mean to avoid that extra package layer and get a more fine grained test output log?
> 
> In the 'develop page', I wrote a note that the structure of unit test should reflect that of the components they test. In this case, the module `conservation_rules` should get a test called `test_conservation_rules` and its functions (say, `identical_particle_symmetrization`), should be named accordingly (`test_identical_particle_symmetrization`). But it's a debatable convention and it's not enforced through style checks or so.

I can see some benefits in that. For example, all of the single function test could be put in the `__init__.py`. That would make the structure nicer. However for some rules, I have more than one test. Each of those tests bundle specific cases together, for example separating fermions and bosons cases. Bundling that altogether again is useful. We could use a class, or just use a module like now. A module seems better in that case.

**Comment by redeboer** [comment]
https://github.com/ComPWA/expertsystem/pull/381#issuecomment-725351302

Earlier on, I tried putting unit tests in the `src` folder, right along the modules that they test, so that there is some sort of enforcement of https://github.com/ComPWA/expertsystem/pull/381#issuecomment-725344704. But this introduces problems like having to filter the tests out when installing the package as well as losing the ability to apply different style checks to the `test_` files, see e.g.
https://github.com/ComPWA/expertsystem/blob/2ef4f442aa62cd18fbf6434acabacd83576048fd/setup.cfg#L123-L126

I would leave it like this, it's more of a thought, but I currently don't see a way how to enforce it automatically.

**Comment by spflueger** [comment]
> For example, all of the single function test could be put in the `__init__.py`.

I just tried this, and it does not work. So I would prefer the current setup.

File: src/expertsystem/reaction/conservation_rules.py (hunk 1 above)
Comment (spflueger) [reply src/expertsystem/reaction/conservation_rules.py]: 6e9f296910162ffcb49c8ec753e5c0ed6f5d0398

**Pull request merged**